### PR TITLE
(LTH-169) Pretty print integers as int64_t

### PR DESCRIPTION
**Pull request by ekinanp**
Signed-off-by: Enis Inan <enis.inan@puppet.com>

### CODE REVIEW

**Comment by ekinanp** [comment]
I tested the `facter_task` out with the changes here and everything worked fine (the `facter_task` previously failed). Here's an example:
```
[root@dh9ah22skdl2a1m ~]# /opt/puppetlabs/bin/puppet-task run facter_task --nodes j41veu0xy1ifsjo.delivery.puppetlabs.net
Starting job ...
Note: The task will run only on permitted nodes.
New job ID: 10
Nodes: 1

Started on j41veu0xy1ifsjo.delivery.puppetlabs.net ...
Finished on node j41veu0xy1ifsjo.delivery.puppetlabs.net
  staging_http_get : curl
  path : /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin
  identity :
    gid : 0
    uid : 0
    user : root
    group : root
    privileged : true

  timezone : UTC
  facterversion : 3.14.5
  dmi :
    bios :
      vendor : Phoenix Technologies LTD
      version : 6.00
      release_date : 12/12/2018

    board :
      product : 440BX Desktop Reference Platform
      manufacturer : Intel Corporation
      serial_number : None

    chassis :
      type : Other
      asset_tag : No Asset Tag

    product :
      name : VMware Virtual Platform
      uuid : 58d71a42-cfd0-cf4e-3255-63f0d0d2e543
      serial_number : VMware-42 1a d7 58 d0 cf 4e cf-32 55 63 f0 d0 d2 e5 43

    manufacturer : VMware, Inc.

  networking :
    interfaces :
      lo :
        ip : 127.0.0.1
        bindings6 : [{"address":"::1","netmask":"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff","network":"::1"}]
        mtu : 65536
        bindings : [{"address":"127.0.0.1","netmask":"255.0.0.0","network":"127.0.0.0"}]
        network6 : ::1
        netmask6 : ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
        ip6 : ::1
        netmask : 255.0.0.0
        network : 127.0.0.0

      ens33 :
        ip : 10.16.121.198
        bindings6 : [{"address":"fe80::250:56ff:fe9a:5b5f","netmask":"ffff:ffff:ffff:ffff::","network":"fe80::"}]
        mtu : 1500
        bindings : [{"address":"10.16.121.198","netmask":"255.255.240.0","network":"10.16.112.0"}]
        network6 : fe80::
        dhcp : 10.32.22.9
        netmask6 : ffff:ffff:ffff:ffff::
        ip6 : fe80::250:56ff:fe9a:5b5f
        netmask : 255.255.240.0
        network : 10.16.112.0
        mac : 00:50:56:9a:5b:5f


    ip : 10.16.121.198
    primary : ens33
    mtu : 1500
    network6 : fe80::
    hostname : j41veu0xy1ifsjo
    dhcp : 10.32.22.9
    fqdn : j41veu0xy1ifsjo.delivery.puppetlabs.net
    netmask6 : ffff:ffff:ffff:ffff::
    ip6 : fe80::250:56ff:fe9a:5b5f
    netmask : 255.255.240.0
    network : 10.16.112.0
    domain : delivery.puppetlabs.net
    mac : 00:50:56:9a:5b:5f

  filesystems : xfs
  processors :
    isa : x86_64
    count : 2
    models : ["Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz","Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz"]
    physicalcount : 2

  is_virtual : true
  should_install_shiro_ini : true
  puppet_files_dir_present : false
  kernelrelease : 4.18.0-80.1.2.el8_0.x86_64
  kernelversion : 4.18.0
  platform_tag : el-8-x86_64
  kernelmajversion : 4.18
  mountpoints :
    /run :
      filesystem : tmpfs
      capacity : 0.85%
      size : 2.83 GiB
      available_bytes : 3013988352
      device : tmpfs
      options : ["rw","nosuid","nodev","mode=755"]
      available : 2.81 GiB
      used_bytes : 25755648
      size_bytes : 3039744000
      used : 24.56 MiB

    /sys/fs/cgroup :
      filesystem : tmpfs
      capacity : 0%
      size : 2.83 GiB
      available_bytes : 3039744000
      device : tmpfs
      options : ["ro","nosuid","nodev","noexec","mode=755"]
      available : 2.83 GiB
      used_bytes : 0
      size_bytes : 3039744000
      used : 0 bytes

    /dev/mqueue :
      filesystem : mqueue
      capacity : 100%
      size : 0 bytes
      available_bytes : 0
      device : mqueue
      options : ["rw","relatime"]
      available : 0 bytes
      used_bytes : 0
      size_bytes : 0
      used : 0 bytes

    /boot :
      filesystem : xfs
      capacity : 21.24%
      size : 1014.00 MiB
      available_bytes : 837369856
      device : /dev/sda1
      options : ["rw","relatime","attr2","inode64","noquota"]
      available : 798.58 MiB
      used_bytes : 225886208
      size_bytes : 1063256064
      used : 215.42 MiB

    / :
      filesystem : xfs
      capacity : 11.02%
      size : 16.99 GiB
      available_bytes : 16228454400
      device : /dev/mapper/rhel-root
      options : ["rw","relatime","attr2","inode64","noquota"]
      available : 15.11 GiB
      used_bytes : 2010476544
      size_bytes : 18238930944
      used : 1.87 GiB

    /dev/hugepages :
      filesystem : hugetlbfs
      capacity : 100%
      size : 0 bytes
      available_bytes : 0
      device : hugetlbfs
      options : ["rw","relatime","pagesize=2M"]
      available : 0 bytes
      used_bytes : 0
      size_bytes : 0
      used : 0 bytes

    /dev/shm :
      filesystem : tmpfs
      capacity : 0%
      size : 2.83 GiB
      available_bytes : 3039744000
      device : tmpfs
      options : ["rw","nosuid","nodev"]
      available : 2.83 GiB
      used_bytes : 0
      size_bytes : 3039744000
      used : 0 bytes

    /dev/pts :
      filesystem : devpts
      capacity : 100%
      size : 0 bytes
      available_bytes : 0
      device : devpts
      options : ["rw","nosuid","noexec","relatime","gid=5","mode=620","ptmxmode=000"]
      available : 0 bytes
      used_bytes : 0
      size_bytes : 0
      used : 0 bytes

    /var/lib/nfs/rpc_pipefs :
      filesystem : rpc_pipefs
      capacity : 100%
      size : 0 bytes
      available_bytes : 0
      device : sunrpc
      options : ["rw","relatime"]
      available : 0 bytes
      used_bytes : 0
      size_bytes : 0
      used : 0 bytes

    /dev :
      filesystem : devtmpfs
      capacity : 0%
      size : 2.82 GiB
      available_bytes : 3025141760
      device : devtmpfs
      options : ["rw","nosuid","size=2954240k","nr_inodes=738560","mode=755"]
      available : 2.82 GiB
      used_bytes : 0
      size_bytes : 3025141760
      used : 0 bytes


  fips_enabled : false
  memory :
    swap :
      used : 0 bytes
      total : 2.00 GiB
      capacity : 0%
      available : 2.00 GiB
      used_bytes : 0
      total_bytes : 2147479552
      available_bytes : 2147479552

    system :
      used : 313.68 MiB
      total : 5.66 GiB
      capacity : 5.41%
      available : 5.36 GiB
      used_bytes : 328912896
      total_bytes : 6079492096
      available_bytes : 5750579200


  is_pe : false
  platform_symlink_writable : true
  augeas :
    version : 1.12.0

  system_uptime :
    days : 0
    hours : 21
    uptime : 21:24 hours
    seconds : 77045

  ssh :
    rsa :
      key : AAAAB3NzaC1yc2EAAAADAQABAAABAQCRzAyfoEXQBbuXDaNMu++Pm87hDqBJ78Jld0AnGc4p8BfC2GyEoVrNWgdcTK6HNnlZSZXNzvFRzC4bg9RXw09oQGOGHZkx4kV6dp5FL7xkBgyOHCPCp9FPCRRB9AtbN7iq4LJqbNd/1FghOoEC6UivsRRkkL2hFXFsmgKtEyx8YICJTY80k+Y1IXZZGfWhAWVoZvQ7L6wDdwMynM7pm+/XGJNRKC5+sK6w5CK1f+k+y94vTqTkk1PIFAPEl0zZfg0KxTWsgqGMBHr7eSPeZKZfh4nz3Lcnk8TlBR6YEaX2sKSOJdDSR/Ot/18AFt/9y/3wNxDQ9yBDPa5GUolCKGpB
      type : ssh-rsa
      fingerprints :
        sha1 : SSHFP 1 1 604ff26fbb5113c2ed163b988e72c227d8baf3a9
        sha256 : SSHFP 1 2 d6470b8a29f0d17d48a920f1b373f7290523f36cf978b568266efaffc73b1355


    ecdsa :
      key : AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBE+Ja68ZcA5Qra3JuvW2QODVzBTfk0X8HJN614y6Z69dZlF0kk8iCocyEOVHlD0PANXFMnNo4rby8Xih+JTnCx4=
      type : ecdsa-sha2-nistp256
      fingerprints :
        sha1 : SSHFP 3 1 32a5d84b5d32aedf1aac082209b1ba9b26f97c8e
        sha256 : SSHFP 3 2 635121b2a0e52aeb8091b3789e97c7244c28627ae2196f79e67a42bc39f0aa7a


    ed25519 :
      key : AAAAC3NzaC1lZDI1NTE5AAAAIGcebXgYlvWzRcQoEsk9tCc2ASqREt/CwpQK5LXHO53n
      type : ssh-ed25519
      fingerprints :
        sha1 : SSHFP 4 1 0dbf7ad0f11e9e551327dc1499be45324e781d1b
        sha256 : SSHFP 4 2 5936ea0026937198d5d531f9853674c78ee8e80dcd1f1b4b2dd5286e088d1c8d



  pe_razor_server_version : package pe-razor-server is not installed
  puppetversion : 6.10.1
  puppet_inventory_metadata :
    packages :
      collection_enabled : false
      last_collection_time : 0.0s


  disks :
    sda :
      size : 20.00 GiB
      model : Virtual disk
      vendor : VMware
      size_bytes : 21474836480

    sr0 :
      size : 1.00 GiB
      model : VMware IDE CDR00
      vendor : NECVMWar
      size_bytes : 1073741312


  kernel : Linux
  virtual : vmware
  os :
    name : RedHat
    family : RedHat
    release :
      full : 8.0
      major : 8
      minor : 0

    selinux :
      enabled : false

    hardware : x86_64
    architecture : x86_64

  partitions :
    /dev/sda1 :
      size : 1.00 GiB
      uuid : ab527550-79d1-4307-bfe1-36f599d770a4
      mount : /boot
      partuuid : e8fabedd-01
      filesystem : xfs
      size_bytes : 1073741824

    /dev/sda2 :
      size : 19.00 GiB
      uuid : jSs09P-2PCM-2yMW-lx3K-xfde-0B7P-IxYWjY
      partuuid : e8fabedd-02
      filesystem : LVM2_member
      size_bytes : 20400046080

    /dev/mapper/rhel-root :
      size : 17.00 GiB
      uuid : 189dafac-96f0-409a-89da-b0beac7fda41
      mount : /
      filesystem : xfs
      size_bytes : 18249416704

    /dev/mapper/rhel-swap :
      size : 2.00 GiB
      uuid : 556043e6-5d3a-4888-ba5e-f1a092208954
      filesystem : swap
      size_bytes : 2147483648


  load_averages :
    1m : 0.000000
    5m : 0.000000
    15m : 0.000000

  hypervisors :
    vmware :
      version :


  ruby :
    sitedir : /opt/puppetlabs/puppet/lib/ruby/site_ruby/2.5.0
    version : 2.5.7
    platform : x86_64-linux

  aio_agent_build : 6.10.1
  aio_agent_version : 6.10.1
  pe_concat_basedir : /opt/puppetlabs/puppet/cache/pe_concat
|
Job completed. 1/1 nodes succeeded.
Duration: 1 sec
```

File: json_container/src/json_container.cc [hunk 1]
```diff
@@ -376,6 +376,19 @@ namespace leatherman { namespace json_container {
         return value.GetInt();
     }
 
+    template<>
+    int64_t JsonContainer::getValue<>(const json_value& value) const {
```
Comment (smcclellan) [inline json_container/src/json_container.cc]: Is the `int` version of `getValue` above still needed?

Comment (ekinanp) [reply json_container/src/json_container.cc]: Yes, for backwards compatibility. See ticket description.

**Comment by ekinanp** [comment]
@smcclellan Updated.

**Pull request merged**